confidant/services/jwkmanager.py

```diff
@@ -38,10 +34,21 @@ def set_key(self, kid: str, private_key: str,
             self._keys.add(key)
         return kid
 
+    def _get_key(self, kid: str):
+        if kid not in self._pem_cache:
+            # setting either way to avoid further lookups when response is None
+            self._pem_cache[kid] = self._keys.get_key(kid)
+            if self._pem_cache[kid]:
```

Comment:
should we remove the`if` statement?   This reads that `get_key(kid)` can return `None`.    If the pem cache is not set, we should just directly set it to the exported pem.

Reply:
if we remove the if this will throw an exception when trying to call `export_to_pem`. It's set up like this because we want `self._pem_cache[kid]` to return `None` if we have already tried to find that key to prevent keep looking for it on each call if it doesn't exist in `self._keys`

Reply:
got it, so we're caching the result `None` here.